src/MeshKernelApi/MeshKernel.cpp

```diff
@@ -1634,7 +1634,7 @@ namespace meshkernelapi
         {
             if (meshKernelState.count(meshKernelId) == 0)
             {
-                throw std::invalid_argument("MeshKernel: The selected mesh kernel id does not exist.");
+                throw std::invalid_argument("mkernel_refine_curvilineargrid: The selected mesh kernel id does not exist.");
```

Comment:
We currently have MeshKernel everywhere.
If we change this here, we should change it everywhere (I am not sure if its necessary though)

Reply:
I think the error message will be more meaningful. let's say you have to debug this exception. By specifying the name of the function you know where to put the breakpoint exactly.

Reply:
If I have access to a C++ compatible debugger, we don't need this information at all since we have access to the call stack.

The reason I added this prefix at all, is because I don't have that for languages like Python or Julia,

If I convert the exceptions to e.g. Python exceptions, I will add the name of the called function anyways so in that case the name would appear twice in the error message. That's why I'd prefer not to mention the name at the API level